app/api/afdb/library/route.ts

```diff
@@ -0,0 +1,241 @@
+import { NextResponse } from 'next/server';
+import { Chapter, Resource, Topic } from '@/app/types';
+import getFetchConfig from '@/api/fetchConfig';
+import { api } from '@/services/url';
+
+const afdbBaseUrl = api.afdb.baseUrl;
+const afdbBearerToken = process.env.AF_DB_SERVICE_BEARER_TOKEN || '';
+
+const fetchData = async (url: string) => {
+    const response = await fetch(url, getFetchConfig(afdbBearerToken));
+
+    if (!response.ok) {
+        throw new Error(`Error: ${response.statusText}`);
+    }
+
+    return response.json();
+};
+
+const fetchWithParams = async (endpoint: string, queryParams: URLSearchParams) => {
+    const urlWithParams = `${afdbBaseUrl}/${endpoint}?${queryParams.toString()}`;
+    return fetchData(urlWithParams);
+};
+
+export async function GET(request: Request) {
+    const { searchParams } = new URL(request.url);
+    const action = searchParams.get('action');
+
+    try {
+        switch (action) {
+            case 'curriculum': {
```

Comment:
These strings are going to be reused no? Maybe move them to a shared constants file?